pkg/stanza/fileconsumer/internal/reader/factory.go

```diff
@@ -119,5 +123,28 @@ func (f *Factory) NewReaderFromMetadata(file *os.File, m *Metadata) (r *Reader,
 	for k, v := range attributes {
 		r.FileAttributes[k] = v
 	}
+
+	if len(f.GzipFileSuffix) > 0 && strings.HasSuffix(file.Name(), f.GzipFileSuffix) {
+		var info os.FileInfo
+
+		if info, err = r.file.Stat(); err != nil {
+			return nil, fmt.Errorf("stat: %w", err)
+		}
+		r.readerFunc = func(file *os.File) (io.Reader, error) {
```

Comment:
not 100% sure about this approach yet, as opposed to setting the reader directly in the struct, but due to the `gzip.NewReader()` function returning an `EOF` error if we are already at the end of the file returning `nil, error` here would propagate up to the caller of the Factory, where this could have some unwanted side effects (i.e. the reader not being added to the trackers).

Reply:
you're right - just saw your suggestion in your PR towards this branch and that looks better to me :)